presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestAvroSchemaUrl.java

```diff
@@ -0,0 +1,101 @@
+        }
+    }
+
+    @AfterTestWithContext
+    public void cleanup()
+    {
+        hdfsClient.delete("/user/hive/warehouse/TestAvroSchemaUrl/schemas");
+    }
+
+    @DataProvider
+    public Object[][] avroSchemaLocations()
+    {
+        return new Object[][] {
+                {"file:///docker/volumes/presto-product-tests/avro/original_schema.avsc"}, // mounted in hadoop and presto containers
+                {"hdfs://hadoop-master:9000/user/hive/warehouse/TestAvroSchemaUrl/schemas/original_schema.avsc"},
+                // TODO {"hdfs:///user/hive/warehouse/TestAvroSchemaUrl/schemas/original_schema.avsc"},
+                // TODO {"/user/hive/warehouse/TestAvroSchemaUrl/schemas/original_schema.avsc"}, // `avro.schema.url` can actually be path on HDFS (not URL)
+        };
+    }
+
+    @Test(dataProvider = "avroSchemaLocations", groups = {AVRO, HIVE_CONNECTOR})
+    public void testHiveCreatedTable(String schemaLocation)
+    {
+        onHive().executeQuery("DROP TABLE IF EXISTS test_avro_schema_url_hive");
+        onHive().executeQuery(format("" +
+                        "CREATE TABLE test_avro_schema_url_hive " +
+                        "ROW FORMAT SERDE 'org.apache.hadoop.hive.serde2.avro.AvroSerDe' " +
+                        "STORED AS " +
+                        "INPUTFORMAT 'org.apache.hadoop.hive.ql.io.avro.AvroContainerInputFormat' " +
+                        "OUTPUTFORMAT 'org.apache.hadoop.hive.ql.io.avro.AvroContainerOutputFormat' " +
+                        "TBLPROPERTIES ('avro.schema.url'='%s')",
+                schemaLocation));
+        onHive().executeQuery("INSERT INTO test_avro_schema_url_hive VALUES ('some text', 123042)");
+
+        assertThat(onHive().executeQuery("SELECT * FROM test_avro_schema_url_hive")).containsExactly(row("some text", 123042));
+        assertThat(onPresto().executeQuery("SELECT * FROM test_avro_schema_url_hive")).containsExactly(row("some text", 123042));
+
+        onHive().executeQuery("DROP TABLE test_avro_schema_url_hive");
+    }
+
+    @Test(dataProvider = "avroSchemaLocations", groups = {AVRO, HIVE_CONNECTOR})
+    public void testPrestoCreatedTable(String schemaLocation)
+    {
+        onPresto().executeQuery("DROP TABLE IF EXISTS test_avro_schema_url_presto");
+        onPresto().executeQuery(format("CREATE TABLE test_avro_schema_url_presto (dummy_col VARCHAR) WITH (format='AVRO', avro_schema_url='%s')", schemaLocation));
+        onPresto().executeQuery("INSERT INTO test_avro_schema_url_presto VALUES ('some text', 123042)");
+
+        assertThat(onHive().executeQuery("SELECT * FROM test_avro_schema_url_presto")).containsExactly(row("some text", 123042));
+        assertThat(onPresto().executeQuery("SELECT * FROM test_avro_schema_url_presto")).containsExactly(row("some text", 123042));
+
+        onPresto().executeQuery("DROP TABLE test_avro_schema_url_presto");
+    }
```

Comment:
Do tests differ only with query executor and create table ddl, do you think it is worth to extract common code? It makes sense to me, if I didn't miss anything.

Reply:
that would make them harder to read.

Reply:
This is very subjective. Now you need to read two tests, but with extraction you would need to understand one test to know what two tests are doing.

Anyway, up to you.